packages/react-native/plugin/src/withBreezIOS.ts

```diff
@@ -0,0 +1,115 @@
+import * as fs from "fs";
+import * as path from "path";
+import type { ExpoConfig } from "@expo/config";
+import {
+  withDangerousMod,
+  withEntitlementsPlist,
+  withPodfile,
+  withXcodeProject,
+} from "@expo/config-plugins";
+
+import { addExtension } from "./addExtension";
+import { withTargetPlist } from "./withExtensionInfoPlist";
+import { warnOnce } from "./utils";
+import { withTargetEntitlementsPlist } from "./withExtensionEntitlementsPlist";
+
+const NOTIFICATION_SERVICE_PODS = (targetName: string) => `
+target '${targetName}' do
+  pod 'BreezSDKLiquid'
+  pod 'KeychainAccess'
+end
+`;
+
+type EASAppExtension = {
+  targetName: string;
+  bundleIdentifier: string;
+  entitlements: Record<string, Array<string>>;
+};
+
+export type NotificationServiceExtensionProps = {
+  apiKey: string;
+  keyService: string;
+  mnemonicKeyName: string;
+}
+
+export function withNotificationServiceExtension(config: ExpoConfig, props: NotificationServiceExtensionProps): ExpoConfig {
+  const appExtensions = (config.extra?.eas?.build?.experimental?.ios
+    ?.appExtensions ?? []) as EASAppExtension[];
+
+  // Set all required variables
+
+  if (appExtensions.length !== 1 || appExtensions[0] === undefined) {
+    warnOnce("Missing App Extension Entry. Please refer to: https://docs.expo.dev/build-reference/app-extensions/");
+    return config;
+  }
+
+  const { targetName, bundleIdentifier, entitlements } = appExtensions[0];
+  const appGroup = entitlements["com.apple.security.application-groups"]?.[0];
+
+  if (appGroup === undefined) {
+    warnOnce("Missing Application Groups entitlement. The extension requires at least one App Group");
+    return config;
+  }
+
+  // Copy required extension files
+  config = withDangerousMod(config, [
+    "ios",
+    (config) => {
+      const extensionDir = path.join(
+        config.modRequest.platformProjectRoot,
+        targetName,
+      );
+      if (!fs.existsSync(extensionDir)) {
+        fs.mkdirSync(extensionDir);
+      }
+
+      const swiftSource = path.join(
+        __dirname,
+        "./../../ios/BreezSDKLiquidNotificationService.swift",
```

Comment:
Is it possible to add this to the plugin instead?
i.e. `react-native/plugin/ios/BreezSDKLiquidNotificationService.swift`